CONTRIBUTING.md

```diff
@@ -17,6 +17,142 @@ limitations under the License.
 
 # Contribution Guidelines
 
+Welcome to NIXL! This document provides guidelines for contributing to our modern C++17 project. Please read through these guidelines carefully before submitting your contribution.
+
+## Table of Contents
+
+1. [Getting Started](#getting-started)
+2. [Development Environment](#development-environment)
+3. [Code Standards](#code-standards)
+4. [Contributing Process](#contributing-process)
+   - [Review Process Expectations](#review-process-expectations)
+5. [Plugin Development](#plugin-development)
+6. [Testing Requirements](#testing-requirements)
+7. [Documentation Standards](#documentation-standards)
+8. [Pull Request Guidelines](#pull-request-guidelines)
+9. [Contribution Rules](#contribution-rules)
+10. [Developer Certificate of Origin](#developer-certificate-of-origin)
+
+## Getting Started
+
+Before contributing, please:
+
+1. Review existing issues and PRs to avoid duplicate work
+2. For significant changes, open an issue for discussion before implementation
+3. Familiarize yourself with our code style and project structure
+4. Set up your development environment according to our guidelines
+
+## Development Environment
+
+### Required Tools
+
+- C++17 compatible compiler
+- Meson build system
+- Ninja build tool
+- clang-format
+- Python (for build scripts and testing)
+- Git with DCO sign-off configured
+
+### Building the Project
+
+NIXL uses Meson and Ninja for building:
+
+```bash
+# Configure the build
+meson setup build
+
+# Build the project
+ninja -C build
+
+# Run tests
+ninja -C build test
+```
+
+### Setting Up clang-format
+
+All C++ code must be formatted using the provided `.clang-format` configuration. Code formatting will be automatically checked for conformance in CI, and improperly formatted code will be rejected:
```

Comment:
we shall say that format of unchanged lines should not be updated

Reply:
done